pan-domain-auth-play_2-4-0/src/main/scala/com/gu/pandomainauth/action/Actions.scala

```diff
@@ -115,12 +115,16 @@ trait AuthActions extends PanDomainAuth {
 
     val existingCookie = readCookie(request) // will be populated if this was a re-auth for expired login
 
-    // use legacy cookie in the check for now, even though both are dropped
-    // phase two of the rollout will be to use the assymetric cookie
+    // use both parsers until we complete the transition to only set the assymetric cookie.
     GoogleAuth.validatedUserIdentity(token).map { claimedAuth =>
       val authedUserData = existingCookie match {
         case Some(c) => {
-          val existingAuth = LegacyCookie.parseCookieData(c.value, settings.secret)
+          val existingAuth = try {
+            CookieUtils.parseCookieData(c.value, settings.publicKey)
+          } catch {
+            case e: Exception =>
```

Comment:
I am interested in only catching the exception thrown by parseCookieData
so I am confident that it will do the right thing.
Thanks, now I know about NonFatal